soc: nordic_nrf: move NFCT_PINS_AS_GPIOS to common nRF definitions

ioannisg opened the pull request:
Kconfig symbol NFCT_PINS_AS_GPIOS may be defined for all
Nordic nRF SoCs that have the NFCT hardware, so we move
the option definition in nRF common Kconfig.soc file. We
also correct the help text to reflect that nRF52-based
DKs and nRF5340 DK have different pins for NFC.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

Fixes #32485

anangl commented on soc/arm/nordic_nrf/Kconfig.soc:
```diff
@@ -3,4 +3,19 @@
+         pins to NFCT mode. Doing this requires UICR erase prior to
+         flashing device using the image which has this option disabled.
+
+         NFC pins in nRF52-based DKs: P0.9 and P0.10
+         NFC pins in nRF5340 DK: P0.2 and P0.3
```
These pins assignments are properties of SoCs, not DKs, so I would rather use here "nRF52 Series" and "nRF5340" (or "nRF53 Series").
And the pins should be specified as P0.09, P0.02, and P0.03, as that's how they are referred in product specifications.

ioannisg replied:
ups, right. For nRF52, I copied the previous names, though. Hope this change won't be an issue to anyone :)

ioannisg replied:
@anangl nRF5340 should be better. Possibly also nRF5340 Application core?

anangl commented on soc/arm/nordic_nrf/Kconfig.soc:
```diff
@@ -3,4 +3,19 @@
 # Copyright (c) 2016-2018 Nordic Semiconductor ASA
 # SPDX-License-Identifier: Apache-2.0
 
+config NFCT_PINS_AS_GPIOS
```
Perhaps it would be better to place it after the `source` command, so that this option appears in `menuconfig` after the MCU Selection one (and others from relevant `Kconfig.soc`).

ioannisg replied:
alright :)

alexsven reviewed (approved):
LGTM

ioannisg committed:
soc: nordic_nrf: move NFCT_PINS_AS_GPIOS to common nRF definitions

Kconfig symbol NFCT_PINS_AS_GPIOS may be defined for all
Nordic nRF SoCs that have the NFCT hardware, so we move
the option definition in nRF common Kconfig file. Also,
we correct the help text to reflect that nRF52 series DKs
and nRF5340 DK have different pins for NFC.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>
